cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ZookeeperCluster.java

```diff
@@ -201,6 +222,43 @@ public static ZookeeperCluster fromAssembly(StatefulSet ss,
         return zk;
     }
 
+    /**
+     * Manage certificates generation based on those already present in the Secrets
+     *
+     * @param certManager CertManager instance for handling certificates creation
+     * @param secrets The Secrets storing certificates
+     */
+    public void generateCertificates(CertManager certManager, List<Secret> secrets) {
+        log.debug("Generating certificates");
+
+        try {
+            Optional<Secret> internalCAsecret = secrets.stream().filter(s -> s.getMetadata().getName().equals(AbstractAssemblyOperator.INTERNAL_CA_NAME))
+                    .findFirst();
+            if (internalCAsecret.isPresent()) {
+
+                // get the generated CA private key + self-signed certificate for internal communications
+                internalCA = new CertAndKey(
+                        decodeFromSecret(internalCAsecret.get(), "internal-ca.key"),
+                        decodeFromSecret(internalCAsecret.get(), "internal-ca.crt"));
+
+                // recover or generates the private key + certificate for each node
+                Optional<Secret> nodesSecret = secrets.stream().filter(s -> s.getMetadata().getName().equals(ZookeeperCluster.nodesSecretName(cluster)))
+                        .findFirst();
+
+                int replicasSecret = !nodesSecret.isPresent() ? 0 : (nodesSecret.get().getData().size() - 1) / 2;
+
+                log.debug("Internal communication certificates");
+                certs = maybeCopyOrGenerateCerts(certManager, nodesSecret, replicasSecret, internalCA, ZookeeperCluster::zookeeperPodName);
+            } else {
+                throw new NoCertificateSecretException("The internal CA certificate Secret is missing");
```

Comment:
During the reconcile, the `AbstractAssemblyOperator` should do that in the `reconcileCertificate` ... but thinking more we should invalidate all the brokers certs and re-create them because at that point they will not be valid anymore. Right? I'll open an issue for that.

Reply:
Well ... they will be valid ... but not accepted. So yes, we should recreate them (probably a separate PR).

Reply:
Yeah I said "not be valid" because the validation process will fails on the brokers.